device/core/devdoc/device_twin_requirements.md

```diff
@@ -3,146 +3,75 @@
-The `fromDeviceclient` method creates a device twin connection to the given hub and calls the `done` method after the connection is complete.
-
-**SRS_NODE_DEVICE_TWIN_18_002: [** `fromDeviceclient` shall throw `ReferenceError` if the `client` object is falsy **]**
-
-**SRS_NODE_DEVICE_TWIN_18_030: [** `fromDeviceclient` shall throw `ReferenceError` if the `done` argument is falsy **]**
-
-**SRS_NODE_DEVICE_TWIN_18_028: [** if `fromDeviceClient` has previously been called for this client, it shall perform a GET operation and return the same object **]**
-
-**SRS_NODE_DEVICE_TWIN_18_029: [** if `fromDeviceClient` is called with 2 different `client`s, it shall return 2 unique `Twin` objects **]**
-
-**SRS_NODE_DEVICE_TWIN_18_003: [** `fromDeviceClient` shall allocate a new `Twin` object **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_004: [** `fromDeviceClient` shall call `getTwinReceiver` on the protocol object to get a twin receiver. **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_005: [** If the protocol does not contain a `getTwinReceiver` method, `fromDeviceClient` shall throw a `NotImplementedError` error **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_007: [** `fromDeviceClient` shall add handlers for the both the `subscribed` and `error` events on the twinReceiver **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_031: [** `fromDeviceClient` shall subscribe to the response topic **]**
-
-**SRS_NODE_DEVICE_TWIN_18_032: [** `fromDeviceClient` shall subscribe to the desired property patch topic **]**
-
-**SRS_NODE_DEVICE_TWIN_18_009: [** `fromDeviceClient` shall call the `done` callback passing a `TimeoutError` if it has not received all necessary `subscribed` events within `Twin.timeout` milliseconds. **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_010: [** `fromDeviceClient` shall call the `done` callback passing  the first error that is returned from `error` event on the twinReceiver. **]** 
-
-**SRS_NODE_DEVICE_TWIN_18_044: [** `fromDeviceClient` shall do a GET call to retrieve desired properties from the service. **]**
-
-**SRS_NODE_DEVICE_TWIN_18_043: [** If the GET operation fails, `fromDeviceClient` shall call the done method with the error object in the first parameter **]**
-
-**SRS_NODE_DEVICE_TWIN_18_034: [** `fromDeviceClient` shall ignore any PATCH operations that arrive before the GET returns **]**
-
-**SRS_NODE_DEVICE_TWIN_18_012: [** `fromDeviceClient` shall remove the handlers for both the `subscribed` and `error` events before calling the `done` callback. **]**
-
-
-### _sendTwinRequest method (private)
-`_sendTwinRequest` is a helper method which sends an arbitrary message to the hub and waits for a response (or timeout or error)
+### constructor(transport: Client.Transport);
 
-**SRS_NODE_DEVICE_TWIN_18_014: [** `_sendTwinRequest` shall use an arbitrary starting `rid` and increment it by one for every call to `_sendTwinRequest` **]** 
+**SRS_NODE_DEVICE_TWIN_16_001: [** The `Twin` constructor shall subscribe to the `onDesiredPropertiesUpdate` event off the `transport` object. **]**
```

Comment:
>** The `Twin` constructor shall subscribe to the `onDesiredPropertiesUpdate` event off the `transport` object. **] [](start = 32, length = 114)

So delta's are always subscribed to?

Reply:
Good point.  I like Robledo's idea that clients should be able to unsubscribe at will to control bandwidth.  The current implementation goes against that.  From what I've seen, after the client calls getTwin, there's no way to disable it again.  Maybe we need to add the manual enable/disable functions into the client API.

---
In reply to: [171908897](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171908897) [](ancestors = 171908897)

Reply:
yes. I have a comment somewhere in the code calling this out. I think the "usual" way we would go about that would be to account for desired properties listeners, and if there are none, disable the feature. it's not trivial though since many listeners can use different properties paths. I will track this down with a separate task

---
In reply to: [171955709](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171955709) [](ancestors = 171955709,171908897)